blivet/fcoe.py

```diff
@@ -155,30 +204,46 @@ def write(self, root):
         if not self.nics:
             return
 
-        if not os.path.isdir(root + "/etc/fcoe"):
-            os.makedirs(root + "/etc/fcoe", 0o755)
-
-        for nic, dcb, auto_vlan in self.nics:
-            fd = os.open(root + "/etc/fcoe/cfg-" + nic,
-                         os.O_RDWR | os.O_CREAT)
-            config = '# Created by anaconda\n'
-            config += '# Enable/Disable FCoE service at the Ethernet port\n'
-            config += 'FCOE_ENABLE="yes"\n'
-            config += '# Indicate if DCB service is required at the Ethernet port\n'
-            config += 'Ethernet port\n'
-            if dcb:
-                config += 'DCB_REQUIRED="yes"\n'
-            else:
-                config += 'DCB_REQUIRED="no"\n'
-            config += '# Indicate if VLAN discovery should be handled by fcoemon\n'
-            if auto_vlan:
-                config += 'AUTO_VLAN="yes"\n'
-            else:
-                config += 'AUTO_VLAN="no"\n'
-            os.write(fd, config.encode('utf-8'))
-            os.close(fd)
+        # Done before packages are installed so don't call
+        # write_nic_fcoe_cfg in target root but just copy the cfgs
+        shutil.copytree("/etc/fcoe", root + "/etc/fcoe")
 
-        return
+    def write_nic_fcoe_cfg(self, nic, dcb=True, auto_vlan=True, enable=True, mode=None, root=""):
+        cfg_dir = root + "/etc/fcoe"
+        example_cfg = os.path.join(cfg_dir, "cfg-ethx")
+        if os.access(example_cfg, os.R_OK):
+            lines = open(example_cfg, "r").readlines()
+        else:
+            anaconda_cfg = """FCOE_ENABLE="yes"
+DCB_REQUIRED="yes"
+AUTO_VLAN="yes"
+MODE="fabric"
+"""
+            lines = anaconda_cfg.splitlines(True)
+
+        with open(os.path.join(cfg_dir, "cfg-%s" % nic), "w") as new_cfg:
+            new_cfg.write("# Generated by Anaconda installer\n")
```

Comment:
Well, I must try - could we just remove the comment for this patch?

I have troubles figuring out how this code (and the related code, which is the code for activating fcoe targets) should be moved to anaconda. The write_nic_fcoe_cfg is just part of activating an fcoe target in anaconda/blivet in this approach (write config file + resrtart fcoe service). I think the code belongs to blivet as the code in the same way iscsi.py and zfcp.py belong here? Another question is that the approach might not be the right one. So perhaps we should try to change it to be actually (not sure if it is currently is) working in non-installer environment?  Try to use fcoe tools (or libs if they exist) for that instead of the write config + restart fcoe service approach. Seems like quite a big task to be solved separately. The service restart was approved by fcoe savvy people, I was also looking for a way using fcoe-tools (fcoeadm) but didn't succeed at the time I was working on the patch for rhel 7.